server/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java

```diff
@@ -77,6 +88,9 @@ void setUseAdaptiveReplicaSelection(boolean useAdaptiveReplicaSelection) {
     }
 
     private void setAwarenessAttributes(List<String> awarenessAttributes) {
+        if (this.awarenessAttributes.isEmpty() && awarenessAttributes.isEmpty() == false) {
+            deprecationLogger.deprecated(IGNORE_AWARENESS_ATTRIBUTES_DEPRECATION_MESSAGE);
```

Comment:
I think we should suppress this warning if `es.search.ignore_awareness_attributes` is already set to `true`.

Reply:
Good catch, in fact we should check the es.search.ignore_awareness_attributes property again since we want to ignore the attributes if it is set to true. Currently any update to the awareness attributes would make search using them even if the system property is set to true. I pushed 2cb7834 to address this.